configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/ObjectMapper.java

```diff
@@ -132,6 +132,50 @@ public void serializeTo(Object instance, ConfigurationNode node) throws ObjectMa
         }
     }
 
+    /* Field (de)serialization if @Adapter annotation is present*/
+    protected static class AdaptedFieldInfo extends FieldData {
```

Comment:
I'm not sure what to think about this being a separate class. I feel like it might be better to put this logic into `FieldData` and store a copy of the `TypeSerializer` (if added) upon class construction that is used on . That would allow overriding classes to make use of this adapter system without having to consider two field data classes.

I don't know what anyone else thinks? Might be something to think about doing for Configurate 4 instead.

Reply:
see a1798168eb0801e45cf8ca47bec282a51a9e490c no longer relevant